cmake/Install.cmake

```diff
@@ -24,17 +24,21 @@ set(configure_input
 "This file was generated from a corresponding .in file.\
  DO NOT MANUALLY EDIT THIS FILE!")
 
+set(extra_datadir
+    "\${prefix}/${rel_datadir}"
+    CACHE STRING "pkg-config path for datadir")
+
 set(extra_completionsdir
-    /usr/local/share/fish/vendor_completions.d
-    CACHE STRING "Path for extra completions")
+    "\${datadir}/fish/vendor_completions.d"
```

Comment:
This will result in

```fish
set __extra_completionsdir ${datadir}/fish/vendor_completions.d
```

in [share/__fish_build_paths.fish](https://github.com/fish-shell/fish-shell/blob/master/share/__fish_build_paths.fish.in),  which is invalid fish code. For fish, you'd want to use `\$datadir` here (similar for $prefix), which results in

```fish
set __extra_completionsdir $datadir/fish/vendor_completions.d
```

I am not sure if pkg-config can deal with that form. If not, it would have to be adjusted.

Reply:
Thinking further about it, keeping these variables unexpanded will *not* work for __fish_build_paths at all, because they won't be defined when that's executed.

So we would have to compute them for that. I assume computing them would also work for pkg-config?

Reply:
Ah, I'll fix that.

>So we would have to compute them for that. I assume computing them would also work for pkg-config?

No, because if the user installs e.g. Sway in a different prefix, the user doesn't want Sway to try to install anything (completions) outside of this prefix. See https://github.com/swaywm/swaybg/issues/13

Reply:
Wait. What?

So you install bash with a prefix of /usr. It computes a datadir of /usr/share, and a completionsdir of /usr/share/bash-completion/completions.

Now you install sway with a prefix of /usr/local. Do you expect the sway bash completions to go to /usr/local/share/bash-completion/completions, and bash to pick that up? How is bash to know? If it knows /usr/local, how is it supposed to know /banana/?

I feel like I'm missing something quite major here - all of my experience building software is the happy path of making packages to use system-wide, so I just use /usr for everything.

So I would expect "the directory where bash completions go" to be a system-wide thing (plus a different "the directory where your own completions go")

Reply:
> Do you expect the sway bash completions to go to /usr/local/share/bash-completion/completions

Yes. No files outside of the prefix.

> and bash to pick that up? How is bash to know? If it knows /usr/local, how is it supposed to know /banana/?

It's up to the user to configure the system-wide bash copy to pick stuff in their custom prefix. It's the same as `LD_LIBRARY_PATH`, `PKG_CONFIG_PATH` and friends needed when installing a library to a custom prefix.

pkg-config variables like `${prefix}` get automatically replaced with the custom prefix by Meson.

Reply:
Yeah it would be nice to have a simpler model here, but I don't see know how that would work with pkg-config, because pkg-config could use a fish.pc from a different prefix than the one you're installing to?

Reply:
Just to illustrate: so if you install fish to prefix `/fish` and grim to prefix `/grim` with your patch then `grim.fish` will be installed to `/fish/share/fish/vendor_completions.d/` (assuming you told pkg-config about the fish.pc). This seems fine. Currently it would be `/usr/local/share/fish/vendor_completions.d`.